### PR TITLE
Add user feedback

### PR DESCRIPTION
**Pull request by Janpot**
Lay the groundwork for a potential user feedback flow.

https://user-images.githubusercontent.com/2109932/182381735-615f2982-1f4e-4357-b91a-a1e1c9720ac9.mov

### CODE REVIEW

File: packages/toolpad-app/src/toolpad/Toolpad.tsx [hunk 1]
```diff
@@ -58,12 +59,14 @@ export default function Toolpad({ basename }: EditorProps) {
       <Box sx={{ height: '1px', minHeight: '100vh' }}>
         <ErrorBoundary fallbackRender={({ error }) => <FullPageError error={error} />}>
           <React.Suspense fallback={<FullPageLoader />}>
-            <BrowserRouter basename={basename}>
-              <Routes>
-                <Route path="/" element={<Home />} />
-                <Route path="/app/:appId/*" element={<AppWorkspace />} />
-              </Routes>
-            </BrowserRouter>
+            <UserFeedback>
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/Toolpad.tsx:62]: i didn't immediately understand this was a context provider from the component name - maybe we could make it more explicit?

**Comment by prakhargupta1** [comment]
Both interactions look good and well positioned.

A copy change suggestion:
Need a specific feature ? You can open a feature request over at our GitHub repo **or upvote from the existing requests.**

For components, can we somehow take users to a page where they can vote among many components ? If not possible on GitHub then may be Notion ?

**Comment by Janpot** [comment]
> For components, can we somehow take users to a page where they can vote among many components ? If not possible on GitHub then may be Notion ?

We could make a GitHub issue for each component?

**Comment by prakhargupta1** [comment]
> We could make a GitHub issue for each component?
This seems straight forward, but won't it become additional work to manage ? I mean for the scenario when the component wasn't listed there: a user clicks on another component and creates an issue for the component they need, but it won't reflect in this catalog (additional work to show it) and may miss some votes.

But since there are only a certain number of popular components users will request first, we can start with a list of top-10 each linking to a GH issue.
Later on we can have a generic: Request a component that can take users to a GH tag having other components' issues.

**Comment by prakhargupta1** [comment]
Cool, I'll share the list of components.

The top-right position is also suitable for: 
1. linking documentation 
2. showing current version of Toolpad

Should we have a small menu there or where else can we place these? @gerdadesign

**Comment by gerdadesign** [comment]
Screen Shot 2022-08-05 at 11 13 00 AM
Is something like this what you're looking for? 
- other icons in top bar appear to be 100% black at 24px square
- tooltip (these appear to be grey in other places in toolpad) that says "Help and resources" 
- would "Submit feedback" link out to this aforementioned singular github issue or to https://github.com/mui/mui-toolpad/issues ?

**Comment by Janpot** [comment]
> * do you think editor and releases can be collapsed into this menu?

It's possible, we should probably take a look at app navigation as a whole as well, but I would leave it out of the scope of this PR.

> * would "Submit feedback" link out to this aforementioned singular github issue or to https://github.com/mui/mui-toolpad/issues ?

either https://github.com/mui/mui-toolpad/issues or https://github.com/mui/mui-toolpad/issues/new?assignees=&labels=status%3A+needs+triage&template=2.feature.yml

**Comment by prakhargupta1** [comment]
Isn't this Help icon more suited in the top right corner ?

Also, instead of Submit feedback we can have:
1. Report bug: https://github.com/mui/mui-toolpad/issues/new?assignees=&labels=status%3A+needs+triage&template=1.bug.yml

2. Request or upvote feature: https://github.com/mui/mui-toolpad/issues/new?assignees=&labels=status%3A+needs+triage&template=2.feature.yml

Rest everything else looks good.
Another out of PR thought: Shouldn't the EDITOR be shown selected when the user is on Editor tab ?

**Comment by gerdadesign** [comment]
> Isn't this Help icon more suited in the top right corner ?

That was my initial instinct as well, but it felt a little confusing sitting next to App deployment functions — the help icon isn't directly related to Previewing the app, so I moved it. In general though I agree that overall navigational elements could be revisited. 

> Request or upvote feature: https://github.com/mui/mui-toolpad/issues/new?assignees=&labels=status%3A+needs+triage&template=2.feature.yml

This link goes directly to Request feature, so to add the option to upvote seems like it should be a step back to https://github.com/mui/mui-toolpad/issues ?


Screen Shot 2022-08-05 at 12 34 23 PM

**Comment by prakhargupta1** [comment]
> In general though I agree that overall navigational elements could be revisited.
Agree, let's go ahead with what you shared.

> This link goes directly to Request feature, so to add the option to upvote seems like it should be a step back to https://github.com/mui/mui-toolpad/issues ?
Agree, let's do this.

**Comment by oliviertassinari** [comment]
A quick benchmark on the support help:

| Product | Position     | Close                                                                                                                                                                | Open                                                                                                                                                                 |
|---------|--------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Ahrefs  | Bottom right | Screenshot 2022-08-06 at 14 23 19 | Screenshot 2022-08-06 at 14 23 22 |
| Figma   | Bottom right | Screenshot 2022-08-06 at 14 23 03 | Screenshot 2022-08-06 at 14 23 09 |
| Retool  | Bottom right | Screenshot 2022-08-06 at 14 24 21 | Screenshot 2022-08-06 at 14 24 26 |
| Ashby   | Bottom left  | Screenshot 2022-08-06 at 14 26 08  | Screenshot 2022-08-06 at 14 26 13 |

**Comment by gerdadesign** [comment]
We could certainly pin it to the bottom! Does this get in the way of any functionality?
Screen Shot 2022-08-06 at 9 54 17 AM

**Comment by prakhargupta1** [comment]
More points to note: 

1. If we show it next to Releases tab then during App preview scenario the header bar would be missing => help icon would be missing.
2. In the App board view there is no Editor, Releases tab, and we will have to horizontally shift help icon.

So keeping it at bottom right as an overlay (as shared by Gerda) appears to be the best placement.

**Comment by Janpot** [comment]
> Does this get in the way of any functionality?

Probably yes, e.g. when a component is selected with several properties:

Screenshot 2022-08-08 at 09 38 44

Personally not a big fan of those floating help/chat buttons. They tend to get in the way, you have to design every screen in your application around them. The header is a surface that is present on every page, wouldn't that be the ideal location and where people would naturally look?

There's a bit of a selection bias in the above benchmark, adding a few more:

| Product  | Position | Screen |
| ------------- | ------------- | ---- |
| appsmith  | header | Screenshot 2022-08-08 at 09 46 56 |
| grafana | bottom left |  Screenshot 2022-08-08 at 09 49 36 |
| google sheets | application menu |  Screenshot 2022-08-08 at 09 52 44 |
| gmail | header top right |  Screenshot 2022-08-08 at 09 55 36 |

**Comment by prakhargupta1** [comment]
Ok. So if we place it on the header bar, then it has to be in the right corner for consistency and that's how some other products have done it.

And, it will be missing from the preview view, which should be ok as it is app preview and the entire header bar won't be there.

Screenshot 2022-08-08 at 4 23 37 PM

I think we can close this here.

**Pull request merged**